### PR TITLE
Provide 2 hooks for adding fields to the attribute creation form

### PR DESCRIPTION
**Pull request by Biont**
Instead of hardcoding specific fields as proposed in #11549, let's just allow developers to hook into the form with a hook at the top and bottom of the form. This is cleaner, more flexible and probably should have been there all along. I'm certainly not against adding new markup for fields that are required by WC itself, but that shuts out plugin developers and forces them to use ugly JS-based hacks in order to inject fields

The `woocommerce_attribute_added` hook can then be used to store the information in an option, so there's no further code needed there.

See also #11529 which requests exactly what my PR does, but ends up adding markup as mentioned above

### CODE REVIEW

File: includes/admin/class-wc-admin-attributes.php [hunk 1]
```diff
@@ -440,6 +440,7 @@ public static function add_attribute() {
 							<h2><?php _e( 'Add New Attribute', 'woocommerce' ); ?></h2>
 							<p><?php _e( 'Attributes let you define extra product data, such as size or colour. You can use these attributes in the shop sidebar using the "layered nav" widgets. Please note: you cannot rename an attribute later on.', 'woocommerce' ); ?></p>
 							<form action="edit.php?post_type=product&amp;page=product_attributes" method="post">
+								<?php do_action( 'woocommerce_add_attribute_top' ) ?>
```
Comment (claudiosmweb) [inline includes/admin/class-wc-admin-attributes.php]: Change to `woocommerce_add_attribute_fields_before`

File: includes/admin/class-wc-admin-attributes.php [hunk 2]
```diff
@@ -488,7 +489,7 @@ public static function add_attribute() {
 									</select>
 									<p class="description"><?php _e( 'Determines the sort order of the terms on the frontend shop product pages. If using custom ordering, you can drag and drop the terms in this attribute.', 'woocommerce' ); ?></p>
 								</div>
-
+								<?php do_action( 'woocommerce_add_attribute_bottom' ) ?>
```
Comment (claudiosmweb) [inline includes/admin/class-wc-admin-attributes.php]: Change to `woocommerce_add_attribute_fields_after`

**Comment by nimmolo** [comment]
I like this better (i'm the author of the similar #11529 ). 

Has there been any discussion here of completely re-writing the Attributes as **terms** rather than **taxonomies**, now that WP has `term_meta`? 

Seems to me like this would be a more elegant and permanent solution than all the querying for taxonomies that match the string pattern `"pa_{attribute_name}"`.

The way I see this working is 
`product_att` would become a new **taxonomy**, like `product_cat`
_Attributes_ would become **terms** of `product_att`
_Attribute options_ (currently the _terms_ of each attribute taxonomy) would become **term_meta**

... And then instead of this custom coded Attributes add screen and edit screen, we could just use the taxonomy add and edit screens like `product_cat` does. Further meta like my PR requested could be added and stored as term meta rather than options.

If you guys like this idea I'll work on it and submit a PR. The recode would obviously have to check for existing pa_taxonomies and move them into terms under the 'product_att' tax, and existing terms would have to be moved to meta. Meta box product data would have to use the new stuff too. But ultimately the code would be simpler.

File: includes/admin/class-wc-admin-attributes.php (hunk 1 above)
Comment (nimmolo) [inline includes/admin/class-wc-admin-attributes.php]: Don't these hooks need to be in the Edit Attribute form too? (~lines 272 and 343)

**Comment by Biont** [comment]
@claudiosmweb: I'll rename the hooks first thing on monday, thanks for the feedback.

@nimmolo: I think turning attribute terms into term meta would have major performance implications for larger shops. The current implementation allows for pretty fast tax queries while meta queries have often lead to performance problems. 
http://wordpress.stackexchange.com/questions/48191/how-much-faster-is-a-tax-query-than-a-meta-query

Plus, you'd be breaking existing third-party code. Of course, I am in no position to decide on this, just sharing my thoughts. I would appreciate if this area of WC would receive some more love, though. At least some de-hardcoding. In any case, though, I think we can agree that any more involving change here will take some time to complete. So the 2 hooks added by this PR will at least provide *some* way for developers to get their hands on that form.

**Comment by nimmolo** [comment]
@Biont that sounds right. Though i'm curious if term_meta queries are as slow as post_meta - it's a separate table, so there would have to be a lot of terms. But I don't doubt such tables exist. (:  

See my line comment - maybe you need those hooks in two different functions in the class? Seems like both the add_attribute and the edit_attribute functions should get the hooks.

**Comment by Biont** [comment]
@nimmolo You were right, I have added the hooks in the edit form as well.
@claudiosmweb I renamed the hooks.

**Comment by mikejolley** [comment]
Will just tweak name further before merge. It seems the hooks in core tend to go for `woocommerce_before_{name} / woocommerce_after_{name}` or `woocommerce_{name}_start / woocommerce_{name}_end` so will make this consistent.

**Comment by Phiph** [comment]
This merge is not available in the latest 2.6.12, it also weirdly shows in github as having the actions. However in the downloadable ZIP & the WC Docs do not include this change.

**Comment by claudiosanches** [comment]
@Phiph or master branch is for 2.7.

**Comment by Phiph** [comment]
Ah okay thanks 👍

**Pull request merged**